Sources/MessagingPush/Integration/CioProviderAgnosticAppDelegate.swift

```diff
@@ -111,6 +111,23 @@ open class CioProviderAgnosticAppDelegate: CioAppDelegateType, UNUserNotificatio
 
     // MARK: - UNUserNotificationCenterDelegate
 
+    open func userNotificationCenter(
+        _ center: UNUserNotificationCenter,
+        willPresent notification: UNNotification,
+        withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void
+    ) {
+        if let wrappedNotificationCenterDelegate = wrappedNotificationCenterDelegate,
+           wrappedNotificationCenterDelegate.responds(to: #selector(UNUserNotificationCenterDelegate.userNotificationCenter(_:willPresent:withCompletionHandler:))) {
+            wrappedNotificationCenterDelegate.userNotificationCenter?(center, willPresent: notification, withCompletionHandler: completionHandler)
+        } else {
+            if #available(iOS 14.0, *) {
+                completionHandler([.list, .banner, .badge, .sound])
+            } else {
+                completionHandler([.alert, .badge, .sound])
+            }
```

Comment:
```suggestion
            if let configOptions = config?(), let showPushAppInForeground = configOptions.showPushAppInForeground, !showPushAppInForeground {
                // Don't show notifications if explicitly disabled
                completionHandler([])
            } else {
                completionHandler([.alert, .badge, .sound])
                // Show notification with appropriate presentation options
                logger.debug("CIO: Using default notification presentation (showing in foreground) because app delegate doesn't implement willPresent")
                if #available(iOS 14.0, *) {
                    completionHandler([.list, .banner, .badge, .sound])
                } else {
                    completionHandler([.alert, .badge, .sound])
                }
            }
```

shouldn't it be like this?

Reply:
This is an interesting suggestion.
I started with an idea to deprecate `showPushAppInForeground` as its behaviour was different from standard iOS, and it had a problematic implementation in the swizzling version of notification handling.

But, now that priority is given to backward-compatibility, it does make sense to reuse `showPushAppInForeground` instead of deprecating it.

I'll investigate whether there are any limitations to doing this. If it works, I'll update the code.

Reply:
Fixed.
Thanks for pointing this out. :)